server/controllers/finance/reports/priceList/index.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @overview server/controllers/finance/reports/priceList/index.js
+ *
+ * @description
+ * This file contains code to create a PDF report for a price list
+ *
+ * @requires db
+ * @requires lodash
+ * @requires util
+ * @requires ReportManager
+ */
+const _ = require('lodash');
+const ReportManager = require('../../../../lib/ReportManager');
+const db = require('../../../../lib/db');
+const util = require('../../../../lib/util');
+
+
+const TEMPLATE = './server/controllers/finance/reports/priceList/report.handlebars';
+
+const PDF_OPTIONS = {
+  filename : 'FORM.LABELS.PRICE_LIST',
+};
+
+exports.report = (req, res, next) => {
+  const options = req.query;
+
+  let report;
+
+  _.defaults(options, PDF_OPTIONS);
+
+  // set up the report with report manager
+  try {
+    report = new ReportManager(TEMPLATE, req.session, options);
+
+    const priceListUUID = db.bid(req.params.uuid);
+    return lookupPriceList(priceListUUID).then(data => {
+      const { items } = data;
+
+      items.forEach(item => {
+        if (item.priceListValue) {
+          if (item.is_percentage === 1) {
+            item.price += util.roundDecimal((item.price * item.priceListValue) / 100, 2);
+          } else {
+            item.price = util.roundDecimal(item.priceListValue, 2);
+          }
+        }
+      });
+      // group by inventory group
+      let groups = _.groupBy(items, i => i.groupName);
+
+      // make sure that they keys are sorted in alphabetical order
+      groups = _.mapValues(groups, lines => {
+        _.sortBy(lines, 'label');
+        return lines;
+      });
+      return report.render({ groups });
+    })
+      .then(result => {
+        res.set(result.headers).send(result.report);
+      })
+      .catch(next)
+      .done();
+
+  } catch (e) {
+    return next(e);
+  }
+};
+
+
+function lookupPriceList(uuid) {
+  let _priceList;
+  let sql =
+    `SELECT BUID(uuid) AS uuid, label, description, created_at, updated_at
+    FROM price_list WHERE uuid = ?;`;
+
+  return db.one(sql, [uuid])
```

Comment:
It doesn't look like the second SQL statement (looking up price list items) depends on the first.   That means you could execute them at the same time to speed the query up.  Like this:
```js
const sql = `SELECT BUID(uuid), ....`;
const itemSql = `SELECT BUID(inventory.uuid) ....`;

return Promise.all([
  db.one(sql, [uuid]),
  db.exec(sql, [uuid]),
])
  .then(([row, rows]) => {
    // do something with the result
  });
```